src/platform/plugins/shared/data/server/deprecations/search_sessions.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import type {
+  CoreSetup,
+  DeprecationsDetails,
+  GetDeprecationsContext,
+  RegisterDeprecationsConfig,
+  SavedObjectsFindResult,
+} from '@kbn/core/server';
+import { i18n } from '@kbn/i18n';
+import type { DeprecationDetailsMessage } from '@kbn/core-deprecations-common';
+import { SEARCH_SESSION_TYPE, SearchSessionSavedObjectAttributes } from '../../common';
+
+type SearchSessionAttributes = Pick<
+  SearchSessionSavedObjectAttributes,
+  'name' | 'username' | 'expires'
+>;
+
+export const createSearchSessionsDeprecationsConfig: (
+  core: CoreSetup
+) => RegisterDeprecationsConfig = (core: CoreSetup) => ({
+  getDeprecations: async (context: GetDeprecationsContext): Promise<DeprecationsDetails[]> => {
+    const [coreStart] = await core.getStartServices();
+    const savedObjectsClient = coreStart.savedObjects.getScopedClient(context.request, {
+      includedHiddenTypes: [SEARCH_SESSION_TYPE],
+    });
+    const finder = savedObjectsClient.createPointInTimeFinder<SearchSessionAttributes>({
+      type: 'search-session',
+      perPage: 1000,
+      fields: ['name', 'username', 'expires'],
+      namespaces: ['*'],
+    });
+
+    const searchSessions: Array<SavedObjectsFindResult<SearchSessionAttributes>> = [];
+
+    for await (const response of finder.find()) {
```

Comment:
Do we need a list of all search sessions in the message? Would not it be enough to show only their total number? My concern is that this operation might take a lot of time for some users who had too many search sessions.